src/AutoRest.CSharp/Mgmt/Models/RequestPath.cs

```diff
@@ -79,13 +79,40 @@ internal struct RequestPath : IEquatable<RequestPath>, IReadOnlyList<Segment>
         /// </summary>
         /// <param name="method"></param>
         public RequestPath(RestClientMethod method)
+            : this(method.Request.PathSegments, method.Operation.GetHttpPath())
+        {
+        }
+
+        public static RequestPath FromPathSegments(PathSegment[] pathSegments, string httpPath)
+        {
+            return new RequestPath(pathSegments, httpPath);
+        }
+
+        private RequestPath(PathSegment[] pathSegments, string httpPath)
         {
             int index = 0;
-            var segments = method.Request.PathSegments
+            var segments = pathSegments
                 .SelectMany(pathSegment => ParsePathSegment(pathSegment, ref index))
                 .ToList();
             _segments = CheckByIdPath(segments);
-            SerializedPath = method.Operation.GetHttpPath();
+            SerializedPath = httpPath;
+        }
+
+        private bool? _isExpandable = null;
+        public bool IsExpandable => _isExpandable ??= GetIsExpandable();
```

Comment:
`RequestPath` is a struct, so `GetIsExpandable` will be called for every new copy of this struct.